src/Model/Tag.php

```diff
@@ -714,17 +725,17 @@ public static function setGlobalTrendingHashtags(int $period, int $limit = 10):
 	 *
 	 * @param int $period Period in hours to consider posts
 	 * @param int $limit  Number of returned tags
+	 * @param int $offset  Page offset in results
 	 * @return array
 	 * @throws \Exception
 	 */
-	public static function getLocalTrendingHashtags(int $period, $limit = 10): array
+	public static function getLocalTrendingHashtags(int $period, $limit = 10, int $offset = 0): array
 	{
-		$tags = DI::cache()->get('local_trending_tags-' . $period . '-' . $limit);
-		if (!empty($tags)) {
-			return $tags;
-		} else {
-			return self::setLocalTrendingHashtags($period, $limit);
+		$tags = DI::cache()->get("local_trending_tags-$period");
+		if (empty($tags)) {
+			$tags = self::setLocalTrendingHashtags($period, 1000);
```

Comment:
This value of `1000` can be really devastating for larger systems, I guess.

Reply:
The number of statuses it has to traverse is the same its just a question of how many entries it returns for the cache. Is there a big system we can compare this query versus a smaller one (100 or the original 20)? On my system the difference was milliseconds.

Reply:
I've got a system with 5,000 users, so I can check.